### ISSUE TITLE
Using browser Back button forces logout

### ISSUE THREAD

**Issue by robin-karlsson** [ForgeRock/oidcSessionCheck#9]
As the script is pushing history by using src on the iframe, we're getting signed out when using the session check library. Do you have any recommendations to avoid this behavior?

On a sidenote, we reproduced the same behavior in the ForgeRock admin console.

**Comment by jakefeasel** [comment]
Can you provide more detailed steps to reproduce? So far I have been unable to do so. Please include:

1) Browser and version
2) Starting page (ideally one of the OOTB ForgeRock UIs)
3) Next page loaded that will trigger the logout after hitting "Back"

If you can also include a HAR file that covers the behavior, that would be very helpful.

Thanks

**Comment by MattNohelty** [comment]
@jakefeasel - The same thing happens in the ForgeRock Platform UI.  It's a bit intermittent and I cannot consistently reproduce with the exact same steps every time so maybe there is a timing issue involved.  However, just clicking around in the Native IDM console a bit you can easily reproduce.  For example: 

1. Log into ForgeRock Platform UI as an admin and navigate to Native Consoles -> IDM.  
2. Click Manage and pick any entity that has some data.  
3. When the grid loads, click on one of the rows.  Then hit the back button. 
4. If you are not logged out, repeat step 3 and you will get logged out within the first few tries.

I've seen this happen in the Platform UI as well.

1. Log into ForgeRock Platform UI as an admin and you will land on Dashboards.  
2. Click Applications and then hit the back button.  
3. 4. If you are not logged out, repeat step 2 and you will get logged out within the first few tries.

**Comment by robin-karlsson** [comment]
Thanks Matt, yes that is similar to what we did. When we reported the issue we did like this:

1. Sign in to ForgeRock Platform UI. Make sure the session check runs once (by clicking somewhere)
2. Go to another page. Make sure the session check runs again.
3. Press the browser's back button.

The problem seems to be that the history state in the iframe goes back one step, and therefore is using the "old" nonce from the IdP response and compare it with the later nonce in session storage.
